src/esssans/normalization.py

```diff
@@ -160,38 +130,42 @@ def iofq_denominator(
         The transmission monitor counts from the direct run (depends on wavelength).
     direct_beam:
         The DataArray containing the direct beam function (depends on wavelength).
-    signal_over_monitor_threshold:
-        The threshold for the ratio of detector counts to monitor counts above which
-        an error is raised because it is not safe to drop the variances of the monitor.
+    uncertainties:
+        The mode for broadcasting uncertainties. See
+        :py:class:`UncertaintyBroadcastMode` for details.
 
     Returns
     -------
     :
         The denominator for the SANS I(Q) normalization.
     """  # noqa: E501
-    # TODO
-    # signal_over_monitor_threshold: float = (0.1,)
     denominator = (
         data_transmission_monitor
         * direct_incident_monitor
         / direct_transmission_monitor
     )
-    if direct_beam is not None:
-        denominator = direct_beam * denominator
 
     # We need to remove the variances because the broadcasting operation between
     # solid_angle (pixel-dependent) and monitors (wavelength-dependent) will fail.
-    # We check beforehand that the ratio of sample detector counts to monitor
-    # counts is small
-    # TODO move this into separate provider?
-    # if denominator.variances is not None:
-    #    _verify_normalization_alpha(
-    #        numerator=data.hist(wavelength=denominator.coords['wavelength']),
-    #        denominator=denominator,
-    #        signal_over_monitor_threshold=signal_over_monitor_threshold,
-    #    )
-
-    denominator = solid_angle * sc.values(denominator)
+    if uncertainties == UncertaintyBroadcastMode.drop:
+        if direct_beam is not None:
+            denominator = direct_beam * sc.values(denominator)
```

Comment:
How do we know, in a general case, that we have to strip the variances of the `denominator` and not the ones of the `direct_beam`?
Maybe in this particular example, it doesn't matter because they get stripped again below before multiplying by the solid angle.

I guess then the assumption that is made is that the `solid_angle` has no variances, because if it does, then it might not be obvious that we need to strip variances from denominator and not the ones from the solid angle.
Basically, should we strip variances from both operands in the broadcast, instead of just one of them?

Reply:
> How do we know, in a general case, that we have to strip the variances of the denominator and not the ones of the direct_beam?

Because the direct beam is wavelength dependent and is never broadcasted here?